src/riak_core_apl.erl

```diff
@@ -40,9 +43,12 @@
 -type ring() :: riak_core_ring:riak_core_ring().
 -type preflist() :: [{index(), node()}].
 -type preflist_ann() :: [{{index(), node()}, primary|fallback}].
+-type preflist_with_number_ann() :: [{{partition_id(), node()}, primary|fallback}].
```

Comment:
Maybe add some documentation so we know what the actual range of values will be; not super clear from the name.